### PR TITLE
Upgraded to the API v1.

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

**Comment by tmatsuo** [comment]
@jonparrott 
It's not urgent. Cloud Pub/Sub launched the v1 api, so I'd like to upgrade the samples to v1. Early next week is good enough.

Since v1 is stable, I started to cache the discovery doc.

TODO: I will make the discovery_doc module as an independent library and upload to pypi later, then start using that library in these samples too.

File: appengine-push/discovery_doc.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Utility for caching the discovery doc.
+
+TODO: Make it an independent library and upload it to pypi.
```
Comment (jonparrott) [inline appengine-push/discovery_doc.py]: Is it worth putting this on pypi if it's a GAE-only module?

File: cmdline-pull/discovery_doc.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
```
Comment (jonparrott) [inline cmdline-pull/discovery_doc.py]: I'm not a fan of having two files named the same thing that do mostly the same thing. Can you consolidate this into one file and make the caching mechanism a parameter ('datastore', 'file')?

**Comment by jonparrott** [comment]
I'm curious as to the value of using the REST API directly from python when gcloud-python exists?

File: appengine-push/discovery_doc.py (hunk 1 above)
Comment (tmatsuo) [inline appengine-push/discovery_doc.py]: My plan is to consolidate the two and upload it to pypi.

File: cmdline-pull/discovery_doc.py (hunk 2 above)
Comment (tmatsuo) [inline cmdline-pull/discovery_doc.py]: Yes that's my plan.

Well, I intended to commit this first and move to the pypi module later, but I would abandon this PR and try creating the pypi module first.

**Comment by tmatsuo** [comment]
It's just a matter of timing because gcloud-python is still WIP.

**Comment by jonparrott** [comment]
gotcha, in that case, LGTM.

File: cmdline-pull/discovery_doc.py (hunk 2 above)
Comment (jonparrott) [inline cmdline-pull/discovery_doc.py]: No need as that'll likely take a while. You can merge this in the meantime.

**Comment by tmatsuo** [comment]
Thanks much! Merging then :)

**Pull request merged**